cpp/Platform.Random/RandomExtensions.h

```diff
@@ -1,11 +1,11 @@
 ﻿namespace Platform::Random
 {
-    class RandomExtensions
-    {
-        public: static std::uint64_t NextUInt64(System::Random random) { return random.NextUInt64(Range.std::uint64_t); }
+    static std::uint64_t NextUInt64(std::mt19937_64& random) { return NextUInt64(random, Ranges::UInt64); }
 
-        public: static std::uint64_t NextUInt64(System::Random random, Range<std::uint64_t> range) { return (std::uint64_t)(random.NextDouble() * range.Difference()) + range.Minimum; }
+    static std::uint64_t NextUInt64(std::mt19937_64& random, const Range<std::uint64_t>& range)
+    {
+        return std::uniform_int_distribution<std::uint64_t>{range.Minimum, range.Maximum}(random);
+    } // I can't really place this function on a single line, because it would seem bloated
```

Comment:
@Konard like it

look at the [real function](https://github.com/linksplatform/Collections/blob/18d85c863d98c91f24d21c22683796100c8a9663/cpp/Platform.Collections/Arrays/GenericArrayExtensions.h#L57-L58)